hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/SparkAvroPostProcessor.java

```diff
@@ -40,8 +40,8 @@ public SparkAvroPostProcessor(TypedProperties props, JavaSparkContext jssc) {
 
   @Override
   public Schema processSchema(Schema schema) {
-    return AvroConversionUtils.convertStructTypeToAvroSchema(
+    return schema != null ? AvroConversionUtils.convertStructTypeToAvroSchema(
```

Comment:
@n3nash : wanted to bring to your notice on this change. Prior to this diff, looks like w/ schema post processor enabled, one can never set target schema to null bcoz, the post processor will try to invoke this call. So, if I am not wrong, the code path we have in DeltaSync.readFromSource(), where we check if userProvidedTargetSchema is null will never be invoked only(bcoz, always the target schema will be non null).  Can you confirm if my understanding is right. 
If yes, may be when we introduced the post processor we missed this flow.

Reply:
I mean, not providing any schema provider at all is supported. This is the case where someone wants to set the source schema but wants to set target schema to null bcoz, they have a transformer and so they do not know the actual schema.

Reply:
@nsivabalan I'm confused now. Let's sync f2f

Reply:
null?